sys/kern/signal.c

```diff
@@ -200,7 +235,10 @@ void sig_kill(proc_t *p, signo_t sig) {
       /* If the thread is sleeping interruptibly (!), wake it up, so that it
        * continues execution and the signal gets delivered soon. */
       if (td_is_interruptible(td)) {
-        sleepq_abort(td);
+        /* XXX Maybe TDF_NEEDSIGCHK should be protected by a different lock? */
+        spin_unlock(&td->td_spin);
```

Comment:
Ok. I see... there's a problem with transferring lock ownership. That won't be fixed till summer :(